### ISSUE TITLE
A way to Iterate though an iterable, without Julia-ising the contents

### ISSUE THREAD

**Issue by oxinabox** [JuliaPy/PyCall.jl#153]
I am trying to work out how to iterate though an python iterable, without the contents being converted to Julia types -- i need them to stay as PyObjects.

I have now created a wrapper to do so:
it is my first time using julia's iterators, and its pretty rough.
I'm also not sure if there isn't already a way to do it (though I've been looking for one for quiet a while).

```
#state = inner_inst, next
immutable PyObjectIter
    inner::PyObject 

    function PyObjectIter(obj::PyObject)
        inner = pycall(obj["__iter__"],PyObject)
        new(inner)
    end
end
function next_or_nothing(pyiter::PyObject)
    try
        pycall(pyiter["next"],PyObject)
    catch PyError
        nothing
    end
end

function Base.start(iter::PyObjectIter)
    inner_inst = deepcopy(iter.inner)
    next = next_or_nothing(inner_inst)
    inner_inst, next
end

function Base.next(pyiter::PyObjectIter, state::(PyObject,Union(Nothing,PyObject)))
    inner_inst, next = state
    new_next = next_or_nothing(inner_inst)
    next, (inner_inst,new_next)
end

function Base.done(pyiter::PyObjectIter, state::(PyObject,Union(Nothing,PyObject)))
    inner_inst, next = state
    next==nothing
end
```

**Comment by stevengj** [comment]
You shouldn't need `deepcopy` etc.  Basically, you just need to follow the [existing PyObject iterator](https://github.com/stevengj/PyCall.jl/blob/0f53e5113184556085f7980c26020316b0793e38/src/conversions.jl#L855-L880) but get rid of the conversion to `PyAny`.

**Comment by tkf** [comment]
closed by #594
